### PR TITLE
Organize Travis jobs by stages to support Travis API builds

### PR DESCRIPTION
**Pull request by jfkhoury**
## Description

- As part of the release process, we want Konductor to trigger our E2E testing suite via an API call to Travis to confirm that the tests still pass before releasing.
- We would like to add a new combination of testing: EDGE `in` vs ALLOY `prod`; this way Konductor can confirm that their integration changes will pass once released to production.
- Support passing the `ALLOY_ENV` env variable in the api call, and remove it from the travis yml.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "functional:local:headless": "EDGE_ENV=\"int\" testcafe chrome:headless",
     "functional:local:build": "EDGE_ENV=\"int\" npm run build:prod && testcafe chrome",
     "functional:local:watch": "EDGE_ENV=\"int\" testcafe chrome --live",
+    "functional:api": "testcafe chrome:headless",
```
Comment (jfkhoury) [inline package.json:27]: TODO: Test on more browsers.

**Comment by jfkhoury** [comment]
@raresvla

File: .travis.yml [hunk 2]
```diff
@@ -14,13 +14,26 @@ services:
...
+        script:
+        - npm ci
+        - npm run build:prod
+        - npm run functional:ci
+        - npm run functional:ci:bs
+        - npm run test:saucelabs
+      - stage: API/Konductor Build
+        if: type == api
+        env:
+          - ALLOY_ENV="prod"
```
Comment (jfkhoury) [inline .travis.yml:33]: Hopefully we can eventually remove this from here and Konductor can pass it in the API. Wasn't able to make it work yet.

**Comment by jonsnyder** [comment]
What happens if we add a functional test in Alloy which relies on a new Alloy feature. That test is going to break if we run it against the latest release of Alloy.  It seems like we should be tagging releases in git, then in travis we can checkout old release tags and run their integration tests against the latest Konductor.

**Comment by jfkhoury** [comment]
@raresvla @Aaronius @jonsnyder   The commit before last solves the issue with the hardcoded ENV Variable in travis.yml, and allows passing it in the API request. Now Konductor can mix and match ENV Variables. Our testing framework supports 2 of them:

`ALLOY_ENV`: Which Alloy to use: `int` or the latest `prod` released version in Akamai (DEFAULT -> INT)
`EDGE_ENV`: Which Konductor Edge Env to use. `int` or `prod`. (DEFAULT -> INT)

**For now**, if you want to test against PROD Alloy, please specify the following branch in the API call: `alloy-latest-release`.

Here's an example request that set Alloy to PROD:

```
body='{
"request": {
"branch":"alloy-latest-release",
"config": {
    "env": {
        "global": ["ALLOY_ENV=prod"]
    }
}
}}'

curl -s -X POST \
   -H "Content-Type: application/json" \
   -H "User-Agent: API Explorer" \
   -H "Accept: application/json" \
   -H "Travis-API-Version: 3" \
   -H "Authorization: token XX" \
   -d "$body" \
   https://api.travis-ci.com/repo/adobe%2Falloy/requests
```
